lisa/sut_orchestrator/azure/features.py

```diff
@@ -145,6 +274,116 @@ def _get_console_log(self, saved_path: Optional[Path]) -> bytes:
 
         return log_response.content
 
+    def _get_connection_string(self) -> str:
+        # setup connection string
+        platform: AzurePlatform = self._platform  # type: ignore
+        connection = self._serial_port_operations.connect(
+            resource_group_name=self._resource_group_name,
+            resource_provider_namespace=self.RESOURCE_PROVIDER_NAMESPACE,
+            parent_resource_type=self.PARENT_RESOURCE_TYPE,
+            parent_resource=self._vm_name,
+            serial_port=self._serial_port.name,
+        )
+        token = platform.credential.get_token("").token
+        serial_port_connection_str = (
+            f"{connection.connection_string}?authorization={token}"
+        )
+
+        return serial_port_connection_str
+
+    @staticmethod
+    async def _run_command_async(connection_str: str, command: str) -> str:
```

Comment:
Move it out of the class, instead of marking it static method. That would be clearer.